### ISSUE TITLE
axe-code is not reporting duplicated links on a page (link-name)

### ISSUE THREAD

**Issue by djalmaaraujo** [dequelabs/axe-core#4171]
### Product

axe-core

### Product Version

4.7.2

### Latest Version

- [X] I have tested the issue with the latest version of the product

#### Expectation
axe-core should report duplicated links on the page, based on https://dequeuniversity.com/rules/axe/4.7/link-name.

#### Actual
axe-code is not reporting duplicated links on the page, based on https://dequeuniversity.com/rules/axe/4.7/link-name.

#### How to Reproduce
Code sample: https://codesandbox.io/s/patient-worker-x7shvk?file=/index.html

1. Open https://codesandbox.io/s/patient-worker-x7shvk?file=/index.html
2. Try to open the link directly (https://x7shvk.csb.app/), without all the CodeSandbox UI
3. Run axe DevTools
4. You should get some issues, but no mention of the duplicated links mentioned in https://dequeuniversity.com/rules/axe/4.7/link-name

VoiceOver shows the issue visually:

![IMG_3164](https://github.com/dequelabs/axe-core/assets/3402/e40c2c1c-b4ec-4e76-b708-d272a49b8cf6)

#### Additional context

Here are the screenshots I took from the report in our application:

- axe results
- Voice-over links
- code/visual

![image (1)](https://github.com/dequelabs/axe-core/assets/3402/3c57163e-c4b7-49f5-97eb-d86a37fc35a6)
![image (2)](https://github.com/dequelabs/axe-core/assets/3402/34191705-5a82-4195-aa0c-f48a97b18a75)
![image](https://github.com/dequelabs/axe-core/assets/3402/2e412e0c-fbac-48d9-a941-6ea9b41a8dff)
![test](https://github.com/dequelabs/axe-core/assets/3402/2b03818b-e9ee-42f2-ae51-725aefd34eda)

**Comment by straker** [comment]
Thanks for the issue. So the `link-name` rule does not check if the name of the link is a duplicate. It only checks if the link has an accessible name (regardless of what it is).

[The rule `identical-links-same-purpose`](https://dequeuniversity.com/rules/axe/4.8/identical-links-same-purpose) does however check the name of the links and verifies that each link that uses the same name also points to the same URL. It only returns Needs Review items though so you won't seem them as violations.

**Comment by djalmaaraujo** [comment]
> Thanks for the issue. So the `link-name` rule does not check if the name of the link is a duplicate. It only checks if the link has an accessible name (regardless of what it is).
> 
> [The rule `identical-links-same-purpose`](https://dequeuniversity.com/rules/axe/4.8/identical-links-same-purpose) does however check the name of the links and verifies that each link that uses the same name also points to the same URL. It only returns Needs Review items though so you won't seem them as violations.

Thank you for the fast response. Does the sentence below deserve a review?

"Link text and alternate text for images, when used as links, must be discernible by a screen reader, **must not have a duplicate label**, and must be focusable."

It says "not have a duplicated label". Is that referring to links or inputs?

**Comment by straker** [comment]
Oh you're right. How interesting. I believe that is a doc error on our end as the rule definitely does not check for duplicate labels.

**Comment by djalmaaraujo** [comment]
Awesome @straker! Thanks for working on this and providing fast feedback!

**Comment by straker** [comment]
I agree. [Success Criterion 2.4.4](https://www.w3.org/WAI/WCAG21/Understanding/link-purpose-in-context.html) is intended to ensure that links can be understood from context alone, however axe-core is unable to determine programmatically if any link text would fail that part of the succession criterion. Axe-core does not and cannot fully satisfy most succession criterion due to needing human judgement call for some parts.
